pkg/api/types.go

```diff
@@ -1664,6 +1667,104 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
```

Comment:
What we've discussed so far was that enforcement of constraints by the `SecurityPolicyProvider` can belong on two levels: api server and kubelet.  On both sides the policy can allow or disallow containers that meet the constraints to run either by outright preventing the docker run (kubelet) or rejecting the pod create/mod (api server).  It is likely that the policy would be represented the same on both sides, I haven't thought of a scenario where they would differ.

https://github.com/GoogleCloudPlatform/kubernetes/pull/6287#discussion_r27631298